webviz_subsurface/plugins/_well_completions.py

```diff
@@ -0,0 +1,473 @@
+    """
+    if df.empty:
+        return [0] * len(time_steps), [0] * len(time_steps)
+
+    events, kh_values = [], []
+    event_value, kh_value = 0, 0
+
+    for timestep in time_steps:
+        if timestep in df.DATE.unique():
+            df_timestep = df[df.DATE == timestep]
+            df_timestep_open = df_timestep[df_timestep["OP/SH"] == "OPEN"]
+
+            # if minimum one of the compdats for the zone is OPEN then the zone is considered open
+            event_value = 1 if not df_timestep_open.empty else -1
+            kh_value = df_timestep_open.KH.sum()
+
+        events.append(event_value)
+        kh_values.append(kh_value)
+    return events, kh_values
+
+
+def get_completion_events_and_kh(
+    df: pd.DataFrame, zone_names: list, time_steps: list
+) -> tuple:
+    # pylint: disable=unused-variable
+    """
+    Extracts completion events ad kh values into two lists of lists of lists,
+    one with completions events and one with kh values
+    * Axis 0 is realization
+    * Axis 1 is zone
+    * Axis 2 is time step
+    """
+    compl_events, kh_values = [], []
+    for rname, realdata in df.groupby("REAL"):
+        compl_events_real, kh_real = [], []
+        for zone_name in zone_names:
+            zone_data = realdata[realdata.ZONE == zone_name]
+            compl_events_real_zone, kh_real_zone = get_time_series(
+                zone_data, time_steps
+            )
+            compl_events_real.append(compl_events_real_zone)
+            kh_real.append(kh_real_zone)
+
+        compl_events.append(compl_events_real)
+        kh_values.append(kh_real)
+    return compl_events, kh_values
+
+
+def format_time_series(
+    open_frac: list, shut_frac: list, kh_mean: list, kh_min: list, kh_max: list
+) -> Optional[Dict]:
+    """
+    The functions takes in five lists with values per timestep
+    * fractions of realizations open in this zone
+    * fractions of realizations shut in this zone
+    * kh mean over open realizations
+    * kh min over open realizations
+    * kh max over open realizations
+
+    Returns the data in compact form:
+    {
+        t: [3, 5],
+        open: [0.25, 1.0],
+        shut: [0.75, 0.0],
+        khMean: [500, 1000],
+        khMin: [200, 800],
+        khMean: [600, 1500]
+    }
+    """
+    output: Dict[str, list] = {
+        "t": [],
+        "open": [],
+        "shut": [],
+        "khMean": [],
+        "khMin": [],
+        "khMax": [],
+    }
+    prev_open_val = prev_shut_val = 0
+
+    for (
+        i,
+        (open_val, shut_val, kh_mean_val, kh_min_val, kh_max_val),
+    ) in enumerate(zip(open_frac, shut_frac, kh_mean, kh_min, kh_max)):
+        if open_val != prev_open_val or shut_val != prev_shut_val:
+            output["t"].append(i)
+            output["open"].append(open_val)
+            output["shut"].append(shut_val)
+            output["khMean"].append(kh_mean_val)
+            output["khMin"].append(kh_min_val)
+            output["khMax"].append(kh_max_val)
```

Comment:
```suggestion
    for i, _ in enumerate(open_frac):
        if open_val != prev_open_val or shut_val != prev_shut_val:
            output["t"].append(i)
            output["open"].append(open_frac[i])
            output["shut"].append(shut_frac[i])
            output["khMean"].append(kh_mean[i])
            output["khMin"].append(kh_min[i])
            output["khMax"].append(kh_max[i])
```
This is subjective. Benefit is fewer temporary local variables.

Reply:
Good idea, I have wondered about how to solve that. Very nice, I didn't know about the _ option